src/components/NewActivityList.js

```diff
@@ -95,44 +143,118 @@ class NewActivityList extends Component {
   addChallenge = () => {
     const authStr = 'Basic ' + window.btoa(this.state.togglKey + ':api_token')
     const time = moment().format('YYYY-MM-DD')
-    axios({
-      headers: {
-        Authorization: authStr
-      },
-      url: 'https://toggl.com/reports/api/v2/details',
-      params: {
-        user_agent: this.state.email,
-        workspace_id: this.state.workspace,
-        user_ids: this.state.user_ids,
-        since: time
-      }
-    })
-    .then((response) => {
-      console.log(response)
-      const timeWorkedBefore = response.data.total_grand
-      this.setState({ timeWorkedBefore })
-
-      const { hours, money, how_long } = this.state
-      if ((hours.length > 0) && (money > 0)) {
-        this.challengeSave({hours, money, how_long, timeWorkedBefore})
-      } else {
-        toastr.error('Please enter correct details')
-      }
-    })
-    .catch((error) => {console.log(error)})
+
+    if (this.state.togglKey && this.state.email && this.state.workspace && this.state.user_ids && this.state.hubstaffAuthToken) {
```

Comment:
refactor this
as i understand we have almost the same functions on frontend and backend
why is that so?
it's ok if only frontend uses those functions, but backend uses as well, then why not move them to backend?